File: ClassicsKotlin/app/src/main/java/com/android/tv/classics/jio/store/PrefStore.java
package com.android.tv.classics.jio.store;

import android.content.Context;
import android.content.SharedPreferences;

import com.android.tv.classics.jio.Constants;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

import org.json.JSONObject;

import java.util.HashMap;
import java.util.Map;

public class PrefStore {
    private SharedPreferences sharedPref;
    private Context context;
    private ObjectMapper objectMapper;

    public PrefStore(Context context){
        this.context= context;
        this.sharedPref = context.getSharedPreferences(Constants.preferenceFile,Context.MODE_PRIVATE);
        this.objectMapper = new ObjectMapper(); // Initialize ObjectMapper here
    }

    public void saveMap(String key, Map<String, String> inputMap){
        try {
            String jsonString = objectMapper.writeValueAsString(inputMap);
            saveData(key, jsonString);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public Map<String, String> getMap(String key){
        Map<String, String> outputMap = null;
        try {
            String jsonString = sharedPref.getString(key, (new JSONObject()).toString());
            // Use TypeReference to specify the Map type
            outputMap = objectMapper.readValue(jsonString, new TypeReference<Map<String, String>>() {});
        } catch (Exception e) {
            e.printStackTrace();
        }
        return outputMap;
    }

    public void saveData(String key, String value){
        SharedPreferences.Editor editor = sharedPref.edit();
        editor.putString(key, value);
        editor.apply(); // Use apply() for asynchronous saving
    }

    public String getData(String key){
        return sharedPref.getString(key, null);
    }

    public void saveBoolean(String key, boolean value){
        SharedPreferences.Editor editor = sharedPref.edit();
        editor.putBoolean(key, value);
        editor.apply(); // Use apply() for asynchronous saving
    }

    public boolean getBoolean(String key){
        return sharedPref.getBoolean(key, false);
    }
}

File: ClassicsKotlin/app/src/main/java/com/android/tv/classics/jio/store/PrefStore.kt
package com.android.tv.classics.jio.store

import android.content.Context
import android.content.SharedPreferences
import com.android.tv.classics.jio.Constants
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import org.json.JSONObject

class PrefStore(private val context: Context) {
    private val sharedPref: SharedPreferences = context.getSharedPreferences(Constants.preferenceFile, Context.MODE_PRIVATE)
    private val objectMapper: ObjectMapper = ObjectMapper()

    fun saveMap(key: String, inputMap: Map<String, String>) {
        try {
            val jsonString = objectMapper.writeValueAsString(inputMap)
            saveData(key, jsonString)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun getMap(key: String): Map<String, String>? {
        try {
            val jsonString = sharedPref.getString(key, JSONObject().toString())
            // Use TypeReference to specify the Map type
            return objectMapper.readValue(jsonString, object : TypeReference<Map<String, String>>() {})
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }

    fun saveData(key: String, value: String) {
        sharedPref.edit().apply {
            putString(key, value)
            apply() // Use apply() for asynchronous saving
        }
    }

    fun getData(key: String): String? {
        return sharedPref.getString(key, null)
    }

    fun saveBoolean(key: String, value: Boolean) {
        sharedPref.edit().apply {
            putBoolean(key, value)
            apply() // Use apply() for asynchronous saving
        }
    }

    fun getBoolean(key: String): Boolean {
        return sharedPref.getBoolean(key, false)
    }
}